### PR TITLE
debt - allow extHostSearch in common

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

**Comment by bpasero** [comment]
@roblourens this can be verfied with "yarn web" (see extension.ts).

I noticed https://github.com/microsoft/vscode/issues/84110 as follow up

File: src/vs/workbench/api/common/extHostSearch.ts [hunk 1]
```diff
@@ -3,14 +3,129 @@
...
+		}
+
+		const query = reviveQuery(rawQuery);
+		const engine = this.createTextSearchManager(query, provider);
+		return engine.search(progress => this._proxy.$handleTextMatch(handle, session, progress), token);
+	}
+
+	protected createTextSearchManager(query: ITextQuery, provider: vscode.TextSearchProvider): TextSearchManager {
+		return new TextSearchManager(query, provider, {
+			readdir: resource => Promise.resolve([]), // TODO@rob implement
```
Comment (bpasero) [inline src/vs/workbench/api/common/extHostSearch.ts]: @roblourens this still needs to be done, possibly by using a registered file system provider if any

File: src/vs/workbench/api/common/extHostSearch.ts [hunk 2]
```diff
@@ -3,14 +3,129 @@
...
+			toCanonicalName: encoding => encoding
+		});
+	}
+}
+
+export function reviveQuery<U extends IRawQuery>(rawQuery: U): U extends IRawTextQuery ? ITextQuery : IFileQuery {
+	return {
+		...<any>rawQuery, // TODO@rob ???
```
Comment (bpasero) [inline src/vs/workbench/api/common/extHostSearch.ts]: @roblourens this todo was already there but not sure what it means

File: src/vs/workbench/services/search/common/textSearchManager.ts [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+		const testingPs: Promise<void>[] = [];
+		const progress = {
+			report: (result: TextSearchResult) => {
+				if (!this.validateProviderResult(result)) {
+					return;
+				}
+
+				const hasSibling = folderQuery.folder.scheme === 'file' ?
+					glob.hasSiblingPromiseFn(() => {
+						return this.fileUtils.readdir(resources.dirname(result.uri));
```
Comment (bpasero) [inline src/vs/workbench/services/search/common/textSearchManager.ts]: @roblourens note that this used to be `path.dirname(result.uri.fsPath)`. I changed it to use `URI` instead.

File: src/vs/workbench/services/search/common/textSearchManager.ts [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+					}) :
+					undefined;
+
+				const relativePath = resources.relativePath(folderQuery.folder, result.uri);
```
Comment (bpasero) [inline src/vs/workbench/services/search/common/textSearchManager.ts]: @roblourens note that this used to be `path.relative(result.uri.fsPath)`. I changed it to use `URI` instead.

**Comment by bpasero** [comment]
@roblourens ping

File: extensions/vscode-api-tests/src/extension.ts [hunk 5]
```diff
@@ -796,16 +796,52 @@ export class MemFS implements vscode.FileSystemProvider, vscode.FileSearchProvid
...
+				for (let i = 0; i < lines.length; i++) {
+					const line = lines[i];
+					const index = line.indexOf(query.pattern);
+					if (index !== -1) {
+						progress.report({
+							uri: file,
+							ranges: new vscode.Range(new vscode.Position(i, index), new vscode.Position(i, index + query.pattern.length)),
+							preview: {
+								text: line,
+								matches: new vscode.Range(new vscode.Position(i, index), new vscode.Position(i, index + query.pattern.length))
```
Comment (roblourens) [inline extensions/vscode-api-tests/src/extension.ts]: This is ranges within the preview, so you should just need this:

```suggestion
								matches: new vscode.Range(new vscode.Position(0, index), new vscode.Position(0, index + query.pattern.length))
```

**Review by roblourens — APPROVED** [review]

**Comment by bpasero** [comment]
Thanks

**Pull request merged**